src/SFA.DAS.Commitments.Database/StoredProcedures/CheckForOverlappingEmailsForTable.sql

```diff
@@ -0,0 +1,30 @@
+﻿CREATE PROCEDURE [dbo].[CheckForOverlappingEmailsForTable]
+	@Emails [EmailCheckTable] READONLY,
+	@CohortId BIGINT = NULL
+AS
+BEGIN
+	SELECT 
+		E.RowId,
+		A.Id, 
+		A.FirstName,
+		A.LastName,
+		A.DateOfBirth,
+		A.CommitmentId AS CohortId,
+		A.StartDate,
+		A.EndDate,
+		A.IsApproved,
+		A.Email,
+		dbo.CourseDatesOverlap(A.StartDate, dbo.GetEndDateForOverlapChecks(A.PaymentStatus, A.EndDate, A.StopDate, A.CompletionDate), E.StartDate, E.EndDate) AS OverlapStatus
+	FROM Apprenticeship A
+	JOIN @Emails E ON E.Email = A.Email
+	WHERE 
+		CASE 
+			WHEN @CohortId IS NULL AND A.IsApproved = 1 THEN 1
+			WHEN @CohortId IS NOT NULL AND A.CommitmentId = @CohortId AND A.IsApproved = 0 AND A.StartDate IS NOT NULL AND A.EndDate IS NOT NULL THEN 1
+			WHEN A.IsApproved = 1 THEN 1
+			ELSE 0
+		END = 1
+		AND A.Id != ISNULL(E.ApprenticeshipId,0) 
+		AND A.Email = E.Email 
+		AND dbo.CourseDatesOverlap(A.StartDate, dbo.GetEndDateForOverlapChecks(A.PaymentStatus, A.EndDate, A.StopDate, A.CompletionDate), E.StartDate, E.EndDate) >= 1 
```

Comment:
Can we use the overlapStatus instead of calling the stored procedure again

Reply:
Not sure what you mean, this field returns the OverlapStatus value to the calling code. SQL will only execute the code once because the same call is made in the where clause

Reply:
I think I see what you mean, but I don't think we can reference the field OverlapStatus as it doesn't exist until after the join, but SQL will optimise this